crates/astria-proto/src/lib.rs

```diff
@@ -24,8 +43,26 @@ mod primitive_impls {
     use crate::primitive::v1::Uint128;
     impl From<u128> for Uint128 {
         fn from(primitive: u128) -> Self {
-            let lo = primitive as u64;
-            let hi = (primitive >> 64) as u64;
+            let [
+                h0,
+                h1,
+                h2,
+                h3,
+                h4,
+                h5,
+                h6,
+                h7,
+                l0,
+                l1,
+                l2,
+                l3,
+                l4,
+                l5,
+                l6,
+                l7,
+            ] = primitive.to_be_bytes();
+            let lo = u64::from_be_bytes([l0, l1, l2, l3, l4, l5, l6, l7]);
+            let hi = u64::from_be_bytes([h0, h1, h2, h3, h4, h5, h6, h7]);
```

Comment:
omg, this is an interesting diff, why does clippy prefer this method?

Reply:
So clippy really hates `as` casts, which I kinda agree with. It's better to rely on `From` and `TryFrom` to make sure one does not get accidental precision loss, wrap arounds, etc (especially when refactoring at some point).

However, in this case it flagged a lossy conversion where I performed `u128 as u64`, which basically cuts off the upper 64 bits. That's technically correct and wanted - but I had to make a choice on whether to add an `#[allow(clippy::cast_possible_truncation)]` or to just spell it out explicitly, shuffling the bytes around as needed.

As you can see I went with the verbose but (in my mind) absolutely and totally cannot-be-confused solution. :laughing: